watchtower/wtclient/types.go

```diff
@@ -0,0 +1,53 @@
+package wtclient
+
+import (
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightningnetwork/lnd/keychain"
+	"github.com/lightningnetwork/lnd/watchtower/wtdb"
+)
+
+// Tower represents the info about a watchtower server that a watchtower client
+// needs in order to connect to it.
+type Tower struct {
+	// ID is the unique, db-assigned, identifier for this tower.
+	ID wtdb.TowerID
+
+	// IdentityKey is the public key of the remote node, used to
+	// authenticate the brontide transport.
+	IdentityKey *btcec.PublicKey
+
+	// Addresses is an AddressIterator that can be used to manage the
+	// addresses for this tower.
+	Addresses *AddressIterator
+}
+
+// NewTowerFromDBTower converts a wtdb.Tower, which uses a static address list,
```

Comment:
Please don't use "newX" to describe concepts. The old concept "x" might disappear, leaving you with only "newX" - confusing. The "newX" concept might need to replace by something else in a few years, and we don't want "newX" to be the old concept (and we also don't want "newNewX"). Furthermore, "newX" doesn't tell us anything more about "X".

Aside from that: I don't know a whole not about `go`, but I think the "FromDBTower" part doesn't add a lot of value, as the `func` is already hardwired to `wtdb.Tower` (which already communicates "DBTower").

My suggestion: "ToTowerWithAddressIterator" or "ToDynamicAddressesTower" or something like that?

Reply:
i think it is pretty standard to use `NewX` for the constructor of `X`